### PR TITLE
[#188] [Generate project with Mason] Migrate the "json_serializable.field_rename" setting

### PR DESCRIPTION
**Pull request by manh-t**
- Solves #188 

## What happened 👀

- Add `json_format` variable  and all options to Mason.
- Replace the hardcode `field_rename`'s value with `json_format`'s value.

## Insight 📝

- This is the last part of migration from old Python script.
- We can select a choice from multiple choices.

## Proof Of Work 📹

https://github.com/nimblehq/flutter-templates/assets/60863885/5bfc648c-ec5c-4bdb-8926-ac7f461b8998

```
? Project name? (your_project_name) your_project_name
? Package name? (co.nimblehq.template) co.nimblehq.template
? App name? (Your App Name) Your App Name
? App version? (0.1.0) 0.1.0
? Build number? (1) 1
? Your default Json formatting? pascal
✓ Made brick template (0.9s)
✓ Generated 123 file(s):
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/.DS_Store (identical)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/test_driver/integration_test.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/integration_test/utils/test_util.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/integration_test/real_app_test.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/integration_test/my_home_page_test.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/codecov.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/test/mocks/generate_mocks.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/test/api/repository/credential_repository_test.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.env.sample (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/l10n.yaml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.metadata (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/pubspec.lock (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcworkspace/contents.xcworkspacedata (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Runner-Bridging-Header.h (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@3x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/LaunchImage.imageset/Contents.json (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-76x76@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@1x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@1x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@1x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-83.5x83.5@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@3x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@3x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-60x60@3x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-60x60@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-76x76@1x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@3x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@2x.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Base.lproj/LaunchScreen.storyboard (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Base.lproj/Main.storyboard (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/AppDelegate.swift (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner/Info.plist (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcodeproj/project.pbxproj (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcodeproj/xcshareddata/xcschemes/production.xcscheme (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Runner.xcodeproj/xcshareddata/xcschemes/staging.xcscheme (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Flutter/Debug.xcconfig (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Flutter/Release.xcconfig (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Flutter/AppFrameworkInfo.plist (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/.gitignore (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Gymfile (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Pluginfile (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Constants/Environments.rb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Constants/Constants.rb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Managers/BuildManager.rb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Managers/MatchManager.rb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Managers/DistributionManager.rb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Fastfile (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/fastlane/Matchfile (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Gemfile (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Gemfile.lock (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Podfile (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/ios/Podfile.lock (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/README.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/pubspec.yaml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.gitignore (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/build.gradle (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/staging/res/values/strings.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/profile/AndroidManifest.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/mipmap-mdpi/ic_launcher.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/mipmap-hdpi/ic_launcher.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/drawable/launch_background.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/values-night/styles.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/values/styles.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/values/strings.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/drawable-v21/launch_background.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/AndroidManifest.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/kotlin/.DS_Store (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/main/kotlin/co/nimblehq/template/MainActivity.kt (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/app/src/debug/AndroidManifest.xml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/gradle/wrapper/gradle-wrapper.properties (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/.gitignore (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/build.gradle (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/gradle.properties (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/android/settings.gradle (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/CODEOWNERS (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/android_deploy_staging.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/android_deploy_production_to_playstore.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/test.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/whatsnew/whatsnew-en-US (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/ios_deploy_staging_to_firebase.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/android_deploy_production.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/ios_deploy_to_testflight.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/ios_deploy_to_app_store.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/workflows/bump_version.yml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/PULL_REQUEST_TEMPLATE.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/PULL_REQUEST_TEMPLATE/release_template.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/ISSUE_TEMPLATE/feature_template.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/ISSUE_TEMPLATE/story_template.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/ISSUE_TEMPLATE/bug_template.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/.github/ISSUE_TEMPLATE/chore_template.md (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/l10n/app_th.arb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/l10n/app_en.arb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/l10n/app_vi.arb (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/di/provider/dio_provider.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/di/interceptor/app_interceptor.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/utils/wrappers/permission_wrapper.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/main.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/model/response/user_response.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/env.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/api/repository/credential_repository.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/api/api_service.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/api/exception/network_exceptions.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/usecases/base/base_use_case.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/lib/usecases/base/use_case_result.dart (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/analysis_options.yaml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/build.yaml (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/assets/images/nimble_logo.png (new)
  /Users/tranmanh/Desktop/nimble/github/flutter_templates/your_project_name/assets/fonts/neuzeit.otf (new)
```

### CODE REVIEW

File: bricks/template/brick.yaml [hunk 1]
```diff
@@ -46,3 +46,13 @@ vars:
     description: Your app version
     default: "1"
     prompt: Build number?
+  json_format:
+    type: enum
+    description: Your default Json formatting when making the network request
+    default: snake
+    prompt: Your default Json formatting?
```
Comment (luongvo) [inline bricks/template/brick.yaml:53]: - To make it match other fields, should we remove `Your` from the prompt & description?
- The correct name of this feature should be `Json field renaming`, right? 
  - https://www.youtube.com/watch?v=Y3ogKymvol0

```suggestion
    description: Default Json field renaming with json_serializable (https://pub.dev/packages/json_serializable#build-configuration)
    default: snake
    prompt: Default Json field renaming format?
```

File: bricks/template/brick.yaml [hunk 2]
```diff
@@ -46,3 +46,13 @@ vars:
     description: Your app version
     default: "1"
     prompt: Build number?
+  json_format:
```
Comment (luongvo) [inline bricks/template/brick.yaml:49]: I think the param should be `json_field_rename_format`. `json_format` makes me feel like JSON:API or other formats of JSON body instead.

**Review by luongvo — COMMENTED** [review]
The rest lgtm

**Review by nmint8m — APPROVED** [review]

File: bricks/template/brick.yaml (hunk 1 above)
Comment (manh-t) [reply bricks/template/brick.yaml]: @luongvo Fixed in 54af0f2

File: bricks/template/brick.yaml (hunk 2 above)
Comment (manh-t) [reply bricks/template/brick.yaml]: @luongvo Fixed in 54af0f2

**Review by luongvo — APPROVED** [review]
All good now

**Pull request merged**